packages/runtime-common/matrix-client.ts

```diff
@@ -74,6 +70,10 @@ export class MatrixClient {
   }
 
   async login() {
+    if (this.loggedInDeferred) {
+      return await this.loggedInDeferred.promise;
+    }
+    this.loggedInDeferred = new Deferred();
```

Comment:
now that we are sharing the matrix client for the realm server amongst the realms we need to be smart about logging in such that all the realms can share the work of a single login attempt as opposed to re-running the login.